llvm/lib/Target/X86/X86TargetTransformInfo.cpp

```diff
@@ -5891,14 +5917,21 @@ bool X86TTIImpl::canMacroFuseCmp() {
 }
 
 bool X86TTIImpl::isLegalMaskedLoad(Type *DataTy, Align Alignment) {
+  bool IsSingleElementVector =
+      isa<VectorType>(DataTy) &&
+      cast<FixedVectorType>(DataTy)->getNumElements() == 1;
+  Type *ScalarTy = DataTy->getScalarType();
+
+  if (ST->hasCF() && IsSingleElementVector &&
+      hasConditionalFaultingLoadStoreForType(ScalarTy))
+    return true;
```

Comment:
```
if (ST->hasCF())
  return hasConditionalFaultingLoadStoreForType(ScalarTy));
```

Reply:
No, it's incorrect. The following code would never be run.